build_tools/benchmarks/comparisons/simple_commands.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import os
+from typing import Optional
+
+from common.benchmark_command import *
+from common.benchmark_command_factory import BenchmarkCommandFactory
+
+_DEFAULT_NUM_BENCHMARK_RUNS = 50
+_DEFAULT_NUM_THREADS = 1
+
+
+class Model:
+
+  def __init__(self, model_name: str, function_input: str,
+               input_name: Optional[str], input_layer: Optional[str]):
+    self.model_name = model_name
+    self.function_input = function_input
+    self.input_name = input_name
+    self.input_layer = input_layer
+
+
+class TfliteWrapper(TFLiteBenchmarkCommand):
+  """Specializes the benchmark command to use TFLite."""
+
+  def __init__(self,
+               benchmark_binary: str,
+               model_name: str,
+               model_path: str,
+               input_layer: Optional[str] = None,
+               input_shape: Optional[str] = None,
+               driver: str = "cpu",
+               num_threads: int = _DEFAULT_NUM_THREADS,
+               num_runs: int = _DEFAULT_NUM_BENCHMARK_RUNS,
+               taskset: Optional[str] = None):
+    super().__init__(benchmark_binary,
+                     model_name,
+                     model_path,
+                     num_threads,
+                     num_runs,
+                     taskset=taskset)
+    self.driver = driver
+    if input_layer and input_shape:
+      self.args.append("--input_layer=%s" % input_layer)
+      self.args.append("--input_layer_shape=%s" % input_shape)
+
+
+class IreeWrapper(IreeBenchmarkCommand):
+  """Specializes the benchmark command to use IREE."""
+
+  def __init__(self,
+               benchmark_binary: str,
+               model_name: str,
+               model_path: str,
+               function_input: str,
+               driver: str = "local-task",
+               num_threads: int = _DEFAULT_NUM_THREADS,
+               num_runs: int = _DEFAULT_NUM_BENCHMARK_RUNS,
+               taskset: Optional[str] = None):
+    super().__init__(benchmark_binary,
+                     model_name,
+                     model_path,
+                     num_threads,
+                     num_runs,
+                     taskset=taskset)
+    self.driver = driver
+    self.args.append("--entry_function=main")
+    self.args.append('--function_input="%s"' % function_input)
+
+
+class SimpleCommandFactory(BenchmarkCommandFactory):
+  """Generates `BenchmarkCommand` objects specific to running series of simple models."""
+
+  def __init__(self, base_dir: str):
+    self._models = [
+        Model("mobilenet_v2_1.0_224", "1x224x224x3xf32", None, None),
+        Model("mobilenet_v2_224_1.0_uint8", "1x224x224x3xui8", "input",
+              "1,224,224,3"),
```

Comment:
The init of a class seems like a surprising place to have this list. Can we instead make this get passed in?

Reply:
Let me try that, I'll move the looping to run_benchmark effectively and just have this set.